### PR TITLE
Query UI: Add tenant box

### PR DESCRIPTION
**Pull request by jacobbaungard**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
- Tenant input box added to the Query UI, in order to be able to specify which tenant the query should use. This is especially useful for setups were tenancy enforcement is enabled  (https://github.com/thanos-io/thanos/pull/6756).

tenant-ui

## Verification

- Manually verified that:
  - No tenant header is sent, if textbox is left empty
  - tenant HTTP header is sent if text is not empty
  - The tenant header name changes with the `query.tenant-header` option
  - The tenant input is added to the URL query params so one can link to the identical query
  
## Todo
* ~~Add tenant header to label calls used during autocompletion~~ Can be done in followup once https://github.com/prometheus/prometheus/pull/13118 has been released.

### CODE REVIEW

**Comment by yeya24** [comment]
Overall I think the UI is neat. But I do have one question. I assume this feature is mainly used for testing purpose. Is it correct? 
For production, having users entering arbitrary tenant value seems not safe.

File: pkg/ui/react-app/src/pages/graph/Panel.tsx [hunk 1]
```diff
@@ -536,6 +556,22 @@ class Panel extends Component<PanelProps & PathPrefixProps, PanelState> {
...
+                  width: 'auto',
+                  marginLeft: '10px',
+                  display: 'inline-block',
+                }}
+                id={`tenant=${id}`}
+                type="text"
+                bsSize="sm"
+                onChange={this.handleChangeTenant}
+                placeholder={`${defaultTenant}`}
+                value={options.tenant}
```
Comment (yeya24) [inline pkg/ui/react-app/src/pages/graph/Panel.tsx:573]: Do we need any validation for the tenant value? Or we want to rely on Thanos backend to do it

**Comment by jacobbaungard** [comment]
> Overall I think the UI is neat. But I do have one question. I assume this feature is mainly used for testing purpose. Is it correct?
> For production, having users entering arbitrary tenant value seems not safe.

In general for the tenancy features, Thanos doesn't do any authentication. If someone has access to the query API, there is nothing stopping anyone from setting an arbitrary tenant with the HTTP headers. It's assumed that you will put something in front of the query API to authenticate users and set the correct tenant.

For the UI specifically, at least as implemented here, I think it should not be exposed directly to end-users if tenancy enforcement is in place. If needed in the future, I think it could be possible to implement this in a way which would allow someone to setup a proxy in front of the UI that first authenticate users, and then set the tenant before reaching the UI, but I am not sure if anyone has that usecase.

File: pkg/ui/react-app/src/pages/graph/Panel.tsx (hunk 1 above)
Comment (jacobbaungard) [reply pkg/ui/react-app/src/pages/graph/Panel.tsx:573]: There's currently just the following validation in the backend: https://github.com/thanos-io/thanos/blob/ec6b8b63d64f9570a56bb633f87245e152e5558c/pkg/tenancy/tenancy.go#L38 that seems fine to just do in the backend imo.

I'm not sure if it makes sense to do any more format validation of the tenant? For receive this is also the only thing we have in place.

**Comment by jacobbaungard** [comment]
Rebased on latest main, and also added changelog. Currently the auto-completetion will always use the default tenant, but we need an improvement in the Prometheus library to implement that. I would suggest we can merge this meanwhile (or at latest shortly after https://github.com/thanos-io/thanos/pull/6756 is merged)

**Comment by jacobbaungard** [comment]
Rebased on main and re-generated the react ui as needed following: https://github.com/thanos-io/thanos/pull/6900

**Review by douglascamata — APPROVED** [review]
LGTM :shipit:

**Review by saswatamcode — APPROVED** [review]
LGTM! Let's merge this. Let's make this textbox optional in a follow-up PR, which is tied to enforcement options

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**